senpy/plugins/conversion/centroids.py

```diff
@@ -37,14 +38,22 @@ def __init__(self, info):
     def _forward_conversion(self, original):
         """Sum the VAD value of all categories found."""
         res = Emotion()
+        totalIntensities = defaultdict(float)
         for e in original.onyx__hasEmotion:
             category = e.onyx__hasEmotionCategory
+            intensity = e.get("onyx__hasEmotionIntensity",1)
             if category in self.centroids:
+                totalIntensities[category] += intensity
                 for dim, value in self.centroids[category].items():
                     try:
-                        res[dim] += value
+                        res[dim] += value * intensity
                     except Exception:
-                        res[dim] = value
+                        res[dim] = value * intensity
+        for dim,intensity in totalIntensities.items():
```

Comment:
@balkian this is where the test should have been falling down: `totalIntensities` was populated with `category` keys, but here it's keys are used in `res`, which is populated with `dim` keys. `res` should have been raising `keyError`s..

It's fixed in the current commit, but this one also passed the tests